### ISSUE TITLE
index type and comment information are missing in show index

### ISSUE THREAD

**Issue by jackysp** [pingcap/tidb#3803]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```
DROP TABLE IF EXISTS test;                                               
CREATE TABLE test (                                                      
  id INTEGER                                                             
  );                                                                     
ALTER TABLE test ADD INDEX idx (id) USING BTREE COMMENT 'idx';           
SHOW INDEX FROM test\G
```

2. What did you expect to see?

*************************** 1. row ***************************
        Table: test
   Non_unique: 1
     Key_name: idx
 Seq_in_index: 1
  Column_name: id
    Collation: A
  Cardinality: 0
     Sub_part: NULL
       Packed: NULL
         Null: YES
   Index_type: BTREE
      Comment: 
Index_comment: idx

3. What did you see instead?

*************************** 1. row ***************************
        Table: test
   Non_unique: 1
     Key_name: idx
 Seq_in_index: 1
  Column_name: id
    Collation: utf8_bin
  Cardinality: 0
     Sub_part: NULL
       Packed: NULL
         Null: YES
   Index_type: 
      Comment: 
Index_comment: 

4. What version of TiDB are you using (`tidb-server -V`)?

Git Commit Hash: f187f88040f8a4bd55fbada86fe254214afdb94c
UTC Build Time:  2017-07-19 03:30:10

**Comment by jackysp** [comment]
In addition, 

```
CREATE TABLE IF NOT EXISTS newtest (                                     
  id INTEGER NOT NULL,                                                   
  k INTEGER DEFAULT '0' NOT NULL,                                        
  PRIMARY KEY (id));                                                     
SHOW INDEX FROM newtest\G
```

will get the index type

*************************** 1. row ***************************
        Table: newtest
   Non_unique: 0
     Key_name: PRIMARY
 Seq_in_index: 1
  Column_name: id
    Collation: utf8_bin
  Cardinality: 0
     Sub_part: NULL
       Packed: NULL
         Null: 
   Index_type: BTREE
      Comment: 
Index_comment:

**Comment by jackysp** [comment]
Fixed #3814
